lib/util.js

```diff
@@ -230,17 +230,31 @@ function format(f) {
   return str;
 }
 
-var debugs = {};
-var debugEnviron;
+const debugs = {};
+let debugEnv;
+
+
+const regExpSpecialChars = /[|\\{}()[\]^$+?.]/g;
+
+function stringToRegExp(debugEnv) {
+  // Escape special chars except wildcard first,
+  // then convert debugEnv to regexp
+  debugEnv = debugEnv.replace(regExpSpecialChars, '\\$&')
+      .replace(/\*/g, '.*')
+      .replace(/,/g, '$|^')
+      .toUpperCase();
+
+  return new RegExp(`^${debugEnv}$`, 'i');
+}
 
 function debuglog(set) {
-  if (debugEnviron === undefined) {
-    debugEnviron = new Set(
-      (process.env.NODE_DEBUG || '').split(',').map((s) => s.toUpperCase()));
+  if (debugEnv === undefined) {
+    debugEnv = process.env.NODE_DEBUG || '';
   }
   set = set.toUpperCase();
   if (!debugs[set]) {
-    if (debugEnviron.has(set)) {
+    const regex = stringToRegExp(debugEnv);
```

Comment:
The RegExp should only be calculated once. It has to be done in the `if (debugEnv === undefined) { ... }` check. I personally would actually move that out of the debuglog to be not lazy (meaning the `NODE_DEBUG` env should be checked when util is required and the RegExp should also be created at that point.

E.g.
```js
let debugRegExp
if (process.env.NODE_DEBUG) {
  debugEnv = process.env.NODE_DEBUG.replace(...).replace(...).replace(...).toUpperCase()
  debugRegExp = new RegExp(...)
}

function debuglog(set) { ... }
```

Reply:
Good point, updated.